### PR TITLE
refactoring: use well known LabelHostname constant

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
`kubernetes.io/hostname` is a well-known label that we are using as a plain string. Switch using the constant provided by k8s.

Before this PR:
`kubernetes.io/hostname` is used as a plain string
After this PR:
`LabelHostname` constant is used

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by orelmisan — APPROVED** [review]
Thank you for the cleanup @fossedihelm 

nit: There is one more under `pkg/virt-handler/node-labeller/cpu_plugin_test.go`.

**Review by orelmisan — APPROVED** [review]

**Review by EdDev — APPROVED** [review]
Nice, thank you!

**Comment by fossedihelm** [comment]
@EdDev @orelmisan Fixed wrong constant and rebased.

**Review by orelmisan — APPROVED** [review]
Thank you

File: pkg/virt-handler/node-labeller/cpu_plugin_test.go [hunk 1]
```diff
@@ -362,7 +362,7 @@ var nodeLabels = map[string]string{
 	"hyperv.node.kubevirt.io/tlbflush":                                 "true",
 	"hyperv.node.kubevirt.io/vpindex":                                  "true",
 	"kubernetes.io/arch":                                               "amd64",
-	"kubernetes.io/hostname":                                           "node01",
+	k8sv1.NamespaceDefault:                                             "node01",
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply pkg/virt-handler/node-labeller/cpu_plugin_test.go]: Good catch!

**Review by jean-edouard — COMMENTED** [review]
/approve

**Pull request merged**